aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -540,22 +540,33 @@ impl AptosVM {
         module_storage: &impl AptosModuleStorage,
         status: ExecutionStatus,
     ) -> ExecutionStatus {
+        use ExecutionStatus::*;
         match status {
-            ExecutionStatus::MoveAbort {
-                location: AbortLocation::Module(module),
+            MoveAbort {
+                location: AbortLocation::Module(module_id),
                 code,
                 ..
             } => {
-                let info = self
-                    .extract_module_metadata(module_storage, &module)
+                let info = module_storage
+                    .fetch_module_metadata(module_id.address(), module_id.name())
+                    .ok()
+                    .flatten()
+                    .and_then(|metadata| get_metadata(&metadata))
                     .and_then(|m| m.extract_abort_info(code));
-                ExecutionStatus::MoveAbort {
-                    location: AbortLocation::Module(module),
+                MoveAbort {
+                    location: AbortLocation::Module(module_id),
                     code,
                     info,
                 }
             },
-            _ => status,
```

Comment:
Fixing the fallthrough pattern is great.
Can we get rid of match though, i.e. does if let MoveAbort { ... } = status work?

Reply:
Changed to `if let ... = status { .. } else { .. }`